components/connect-wallet-button.tsx

```diff
@@ -35,10 +33,13 @@ export default function ConnectWalletButton({children = null, asModal = false, f
     if (!state.Service?.active)
       return;
 
-    if (+state.connectedChain?.id === +state.Settings?.requiredChain?.id) {
+    if (
+        +state.connectedChain?.id === +state.Settings?.requiredChain?.id ||
```

Comment:
this can be abbreviated, `(+state.connectedChain?.id || +window?.ethereum?.chainId) === +state.Settings?.requiredChain?.id`